src/SubscriberBase.h

```diff
@@ -83,6 +83,15 @@ class SubscriberBaseT : public Subscriber<T>,
       bool startExecutor = true)
       : ExecutorBase(executor, startExecutor), cancelled_(false) {}
 
+  virtual ~SubscriberBaseT() {
+    if (!cancelled_ && originalSubscription_) {
```

Comment:
Okay, fair enough. I was doing this in unit tests, where I have a `Subscriber` which is never passed to a `ReactiveSocket`—in those circumstances, it's possible for a `Subscription` to go uncancelled, but I guess that's just an artifact of my test and not something that will happen in prod?